src/views/Notifications/Settings/index.tsx

```diff
@@ -290,7 +366,36 @@ const NotificationSettings = () => {
             innerRef={formikRef}
           >
             {(formikProps: FormikProps<NotificationSettingsFormType>) => {
-              const { values, handleSubmit, dirty } = formikProps;
+              const {
+                dirty,
+                handleSubmit,
+                setFieldValue,
+                values
+              } = formikProps;
+
+              type Obj = { [key: string]: any };
+
+              /**
+               * Returns an object containing the keys and values from `obj1` that also exist in `obj2`.
+               *
+               * @param obj1 - The first object to compare.
+               * @param obj2 - The second object to compare.
+               * @returns An object containing the matching keys and values from `obj1`.
+               */
+              const getMatchingKeys = (obj1: Obj, obj2: Obj): Obj =>
+                Object.keys(obj1)
+                  .filter(key =>
+                    Object.prototype.hasOwnProperty.call(obj2, key)
+                  )
+                  .reduce((acc, key) => {
+                    acc[key] = obj1[key];
+                    return acc;
+                  }, {} as Obj);
+
+              const basicNotificationKeys = getMatchingKeys(
```

Comment:
I think the naming of this is misleading, this represents basic notification values which are being used in the form to represent the checkboxes